File: data/r2dbc/r2dbc-flyway-postgres/src/main/java/com/mahmud/r2dbcflywaypostgres/domain/Todo.java
package com.mahmud.r2dbcflywaypostgres.domain;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Table("todos")
public class Todo {
    @Id
    private Long id;
    private String title;
    private String description;
    private Boolean completed = false;
    private LocalDateTime createdAt;
    private Integer priority = 0; // New field
}

File: io/grpc-mongodb-todo/grpc-server/src/main/kotlin/com/mahmud/grpcserver/model/Todo.kt
package com.mahmud.grpcserver.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant

@Document(collection = "todos")
data class Todo(
    @Id
    val id: String? = null,
    val title: String,
    val description: String,
    val completed: Boolean = false,
    val createdAt: Instant = Instant.now(),
    val updatedAt: Instant = Instant.now(),
    val dueDate: Instant? = null
)
